presto-main/src/main/java/com/facebook/presto/operator/scalar/TestTDigest.java

```diff
@@ -0,0 +1,419 @@
+
+        for (int i = 0; i < quantile.length; i++) {
+            assertContinuousWithinBound(quantile[i], STANDARD_ERROR, list, tDigest);
+        }
+    }
+
+    @Test(dataProvider = "compressionFactors")
+    public void testMergeManyLargeNormalDistributions(int compressionFactor)
+    {
+        TDigest tDigest = TDigest.createMergingDigest(compressionFactor);
+        List<Double> list = new ArrayList<>();
+        NormalDistribution normal = new NormalDistribution(500, 20);
+        int digests = 1000;
+        double[] quantile = {0.0001, 0.0200, 0.0300, 0.04000, 0.0500, 0.1000, 0.2000, 0.3000, 0.4000, 0.5000, 0.6000, 0.7000, 0.8000,
+                0.9000, 0.9500, 0.9600, 0.9700, 0.9800, 0.9999};
+
+        for (int k = 0; k < digests; k++) {
+            TDigest current = TDigest.createMergingDigest(compressionFactor);
+            for (int i = 0; i < NUMBER_OF_ENTRIES / digests; i++) {
+                double value = normal.sample();
+                current.add(value);
+                list.add(value);
+            }
+            tDigest.add(current);
+        }
+
+        Collections.sort(list);
+
+        for (int i = 0; i < quantile.length; i++) {
+            assertContinuousWithinBound(quantile[i], STANDARD_ERROR, list, tDigest);
+        }
+    }
+
+    @Test(dataProvider = "compressionFactors")
+    public void testCompareAccuracy(int compressionFactor)
+    {
+        TDigest tDigest = TDigest.createMergingDigest(compressionFactor);
+        QuantileDigest qDigest = new QuantileDigest(STANDARD_ERROR);
+        List<Double> values = new ArrayList<>();
+        NormalDistribution normal = new NormalDistribution(500, 20);
+        UniformRealDistribution uniform = new UniformRealDistribution(0, 1_000_000_000);
+        UniformIntegerDistribution uniformInteger = new UniformIntegerDistribution(0, 1_000_000_000);
+        double[] quantile = {0.0001, 0.0200, 0.0300, 0.04000, 0.0500, 0.1000, 0.2000, 0.3000, 0.4000, 0.5000, 0.6000, 0.7000, 0.8000,
+                0.9000, 0.9500, 0.9600, 0.9700, 0.9800, 0.9999};
+        double[] quantileExtremes = {0.0001, 0.0005, 0.0010, 0.0050, 0.0100, 0.9900, 0.9950, 0.9990, 0.9995, 0.9999};
+
+        for (int i = 0; i < NUMBER_OF_ENTRIES; i++) {
+            double value = normal.sample();
+            qDigest.add((long) value);
+            tDigest.add(value);
+            values.add(value);
+        }
+        int success = 0;
+        int total = 0;
+        Collections.sort(values);
+        for (int i = 0; i < quantile.length; i++) {
+            double errorQuantileT = normal.cumulativeProbability(tDigest.quantile(quantile[i]));
+            errorQuantileT = (double) Math.round(errorQuantileT * 10000d) / 10000d;
+
+            double errorQuantileQ = normal.cumulativeProbability(qDigest.getQuantile(quantile[i]));
+            errorQuantileQ = (double) Math.round(errorQuantileQ * 10000d) / 10000d;
+
+            //System.out.println(quantile[i] + " " + errorQuantileT);
```

Comment:
We need to remove these System.out.println statements

Reply:
These are leftover from testing and playing around with the function. Will delete as soon as we're happy with overall performance and how functions are written.